### PR TITLE
Implement terrain file reading to generate navmeshes in the future

### PR DESCRIPTION
**Pull request by Jettford**
Implement partial reading of the `.raw` file type for implementation along side `.hkx` reading in future for dynamic navmesh genereation.

### CODE REVIEW

File: dNavigation/dTerrain/RawHeightMap.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class RawHeightMap {
+public:
+	RawHeightMap();
+	RawHeightMap(std::ifstream& stream, float height, float width);
+	~RawHeightMap();
+
+	uint32_t m_Unknown1;
+	uint32_t m_Unknown2;
+	uint32_t m_Unknown3;
+	uint32_t m_Unknown4;
```
Comment (aronwk-aaron) [inline dNavigation/dTerrain/RawHeightMap.h:17]: Why are these unknown?

Comment (Jettford) [reply dNavigation/dTerrain/RawHeightMap.h:17]: It is a custom format, we don't know what those values are.

**Review by aronwk-aaron — APPROVED** [review]
LGTM, really looking forward to the directions this is going

File: dNavigation/dTerrain/RawChunk.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "RawHeightMap.h"
+#include "RawMesh.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawChunk.h:7]: This can be replaced with a forward declaration

File: dNavigation/dTerrain/RawChunk.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#include "BinaryIO.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawChunk.h:9]: Don't believe this is necessary in a header file alongside the `<string>` include

File: dNavigation/dTerrain/RawFile.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#include "RawFile.h"
+
+#include "Game.h"
+#include "dLogger.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawFile.cpp:4]: Extra includes here as well

File: dNavigation/dTerrain/RawFile.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "BinaryIO.h"
+
+#include "RawChunk.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawFile.h:8]: Extra includes and can also forward declare RawChunk

File: dNavigation/dTerrain/RawHeightMap.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "BinaryIO.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawHeightMap.h:6]: Extra include

File: dNavigation/dNavMesh.cpp [hunk 7]
```diff
@@ -1,11 +1,15 @@
 #include "dNavMesh.h"
 
+#include "dTerrain/RawFile.h"
```
Comment (EmosewaMC) [inline dNavigation/dNavMesh.cpp:3]: Why is this needing to specify a directory?  Shouldn't this be taken care of with include_directories?

File: dNavigation/dTerrain/RawMesh.h [hunk 8]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include "NiPoint3.h"
```
Comment (EmosewaMC) [inline dNavigation/dTerrain/RawMesh.h:3]: I believe you can forward declare NiPoint3 here as well, but not sure since its in a vector.

**Review by EmosewaMC — COMMENTED** [review]
Minor feedback on includes structure.  Overall code looks fine.

**Comment by Jettford** [comment]
@EmosewaMC Give this another review, I have made all of your requested changes.

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**